src/get-markdown-notes.js

```diff
@@ -1,27 +1,42 @@
 const fs = require("fs");
 const path = require("path");
 
+function toRegExp(value) {
+  if (typeof value === "string") {
+    return new RegExp(`^${value}$`);
+  }
+  return value;
+}
+
+const matches = (filename) => (regExp) => regExp.test(filename);
+const doesNotMatchAny = (regExps) => (filename) =>
+  !regExps.some(matches(filename));
+
 module.exports = (pluginOptions) => {
   let notesDirectory = pluginOptions.notesDirectory || "content/brain/";
   let notesFileExtensions = pluginOptions.notesFileExtensions || [
     ".md",
     ".mdx",
   ];
+  let exclusions =
+    (pluginOptions.exclude && pluginOptions.exclude.map(toRegExp)) || [];
```

Comment:
My reasoning for mapping all the contents of `exclude` to regexps here is that then we can keep that logic separate from the filtering logic later